### ISSUE TITLE
snapshot creation errors if it's larger than 4mb

### ISSUE THREAD

**Issue by nicks** [tilt-dev/tilt#3687]
If you try to send a snapshot larger than 4mb, you hit the GRPC message size limit and get an error

**Comment by nicks** [comment]
related to https://github.com/tilt-dev/tilt/issues/3194, which it seems was not fully fixed

**Comment by nicks** [comment]
here's the error message on the tilt server side:
grpc: received message larger than max (12096681 vs. 4194304)

**Comment by nicks** [comment]
I think eventually we're just going to need to cap the number of logs in a snapshot, 4mb is a lot of logs for a local dev environment, so maybe that's a better place to start

**Comment by andymartin-sch** [comment]
I think we may have started seeing this now that we're using snapshots again.

`Error creating snapshot: posting snapshot failed. status: 500 Internal Server Error, response: error creating snapshot`

**Comment by nicks** [comment]
yep, the error i see on the backend is - grpc: received message larger than max (6775425 vs. 4194304)

**Comment by andymartin-sch** [comment]
I'm not sure if it's the same underlying cause, but I got the same error this morning when creating a snapshot with Tilt v0.18.8:
`Error creating snapshot: posting snapshot failed. status: 500 Internal Server Error, response: error creating snapshot`

**Comment by andymartin-sch** [comment]
thanks @nicks !

**Comment by andymartin-sch** [comment]
we got something like this recently in the form of:
```
  Snapshot: Error creating snapshot: posting snapshot failed. status: 413 Request Entity Too Large, response: snapshot too large. nodes over 1.0MB are: /view/logList/segments: 2.6MB 
```

**Comment by sgreene570** [comment]
The 4 Mb limit seems less reasonable when considering that some users capture snapshots after running `tilt ci`. These snapshots would include all logs produced during a CI run, as opposed to brief logs created during a dev stint.

Any updates on this issue per chance? We'd love to be able to have a higher snapshot size limit, even if that were to cost $$.

**Comment by andymartin-sch** [comment]
We would also pay $$ if this were coupled with an improved onboarding workflow (https://github.com/tilt-dev/tilt/issues/3515)
